frontend/context/app.tsx

```diff
@@ -2,50 +2,58 @@ import React, { createContext, useReducer, ReactNode } from "react";
 
 import { IUser } from "interfaces/user";
 import { IConfig } from "interfaces/config";
+import { ITeam } from "interfaces/team";
 import permissions from "utilities/permissions";
 
 type Props = {
   children: ReactNode;
 };
 
 type InitialStateType = {
-  currentUser: IUser | null;
   config: IConfig | null;
+  currentUser: IUser | null;
+  currentTeam: ITeam | undefined;
   isFreeTier: boolean | undefined;
   isPremiumTier: boolean | undefined;
   isGlobalAdmin: boolean | undefined;
   isGlobalMaintainer: boolean | undefined;
   isGlobalObserver: boolean | undefined;
   isOnGlobalTeam: boolean | undefined;
   isAnyTeamMaintainer: boolean | undefined;
+  isTeamMaintainer: boolean | undefined;
   isOnlyObserver: boolean | undefined;
   setCurrentUser: (user: IUser) => void;
+  setCurrentTeam: (team: ITeam | undefined) => void;
   setConfig: (config: IConfig) => void;
 };
 
 const initialState = {
-  currentUser: null,
   config: null,
+  currentUser: null,
+  currentTeam: undefined,
   isFreeTier: undefined,
   isPremiumTier: undefined,
   isGlobalAdmin: undefined,
   isGlobalMaintainer: undefined,
   isGlobalObserver: undefined,
   isOnGlobalTeam: undefined,
   isAnyTeamMaintainer: undefined,
+  isTeamMaintainer: undefined,
   isOnlyObserver: undefined,
   setCurrentUser: () => null,
+  setCurrentTeam: () => null,
   setConfig: () => null,
 };
 
 const actions = {
   SET_CURRENT_USER: "SET_CURRENT_USER",
+  SET_CURRENT_TEAM: "SET_CURRENT_TEAM",
   SET_CONFIG: "SET_CONFIG",
 };
 
 // helper function - this is run every
 // time currentUser or config is changed
```

Comment:
Should the comment be updated because also now runs when currentTeam changes (if I'm reading correctly)?